src/containers/FormikForm/formikSubjectpageHooks.ts

```diff
@@ -0,0 +1,149 @@
+import { useEffect, useState } from 'react';
+import * as frontpageApi from '../../modules/frontpage/frontpageApi';
+import {
+  transformSubjectFromApiVersion,
+  transformSubjectToApiVersion,
+  getUrnFromId,
+} from '../../util/subjectHelpers';
+import { SubjectpageApiType, SubjectpageEditType } from '../../interfaces';
+import { updateSubjectContentUri } from '../../modules/taxonomy/subjects';
+import { fetchDraft } from '../../modules/draft/draftApi';
+import {
+  fetchResource,
+  queryResources,
+  queryTopics,
+  queryLearningPathResource,
+} from '../../modules/taxonomy/resources';
+import { fetchTopic } from '../../modules/taxonomy/topics';
+import { fetchLearningpath } from '../../modules/learningpath/learningpathApi';
+
+export function useFetchSubjectpageData(
+  subjectId: string,
+  selectedLanguage: string,
+  subjectpageId: string | undefined,
+) {
+  const [subjectpage, setSubjectpage] = useState<SubjectpageEditType>();
+  const [loading, setLoading] = useState(false);
+
+  const fetchSubjectpage = async () => {
+    if (subjectpageId) {
+      setLoading(true);
+      const subjectpage: SubjectpageApiType = await frontpageApi.fetchSubjectpage(
+        subjectpageId,
+        selectedLanguage,
+      );
+      const editorsChoices = await fetchEditorsChoices(
+        subjectpage.editorsChoices,
+      );
+      setSubjectpage(
+        transformSubjectFromApiVersion(
+          subjectpage,
+          subjectId,
+          selectedLanguage,
+          editorsChoices,
+        ),
+      );
+      setLoading(false);
+    }
+  };
+
+  const fetchEditorsChoices = async (resourceUrn: string[]) => {
+    const taxonomyResources = await Promise.all(
+      resourceUrn.map(urn => {
+        if (urn.split(':')[1] === 'topic') {
+          return fetchTopic(urn);
+        }
+        return fetchResource(urn);
+      }),
+    );
+    const articleIds = taxonomyResources.map(resource =>
+      resource.contentUri.split(':'),
+    );
+    return await Promise.all(
+      articleIds.map(async articleId => {
+        if (articleId[1] === 'learningpath') {
+          const learningpath = await fetchLearningpath(articleId.pop());
+          return {
+            ...learningpath,
+            metaImage: {
+              url: learningpath.coverPhoto.url,
+            },
+          };
+        }
+        return fetchDraft(articleId.pop());
+      }),
+    );
+  };
```

Comment:
Kanskje bedre navning for å gjenspeile kva som hentes.
```suggestion
  const fetchEditorsChoices = async (taxonomyUrn: string[]) => {
    const taxonomyElements = await Promise.all(
      taxonomyUrn.map(urn => {
        if (urn.split(':')[1] === 'topic') {
          return fetchTopic(urn);
        }
        return fetchResource(urn);
      }),
    );
    const elementIds = taxonomyElements.map(element =>
      element.contentUri.split(':'),
    );
    return await Promise.all(
      elementIds.map(async elementId => {
        if (elementId[1] === 'learningpath') {
          const learningpath = await fetchLearningpath(elementId.pop());
          return {
            ...learningpath,
            metaImage: {
              url: learningpath.coverPhoto.url,
            },
          };
        }
        return fetchDraft(elementId.pop());
      }),
    );
  };
```